src/components/LocationPermissionModal/index.tsx

```diff
@@ -37,12 +71,24 @@ function LocationPermissionModal({startPermissionFlow, resetPermissionFlow, onDe
 
     const handledBlockedPermission = (cb: () => void) => () => {
         if (hasError) {
+            // native
             if (Linking.openSettings) {
                 Linking.openSettings();
+                setShowModal(false);
+            } else if (isWeb) {
```

Comment:
Hmm we should use separate files for platform specific code, e.g. `LocationPermissionModal/index.native.tsx`, `LocationPermissionModal/index.web.tsx`, `LocationPermissionModal/index.desktop.tsx`

Reply:
@luacmartins since we already have `isWeb`, I thought using the same file would be more efficient to minimize changes, but let me know if we need to make it separated

Reply:
the file now separated